### ISSUE TITLE
Crewmembers with identical names share the same Medical/Security records when examined via a HUD

### ISSUE THREAD

**Issue by atosti** [tgstation/tgstation#73963]
## Round ID:
N/A. Tested locally on the latest tgstation master branch. (Commit: 1f5391817df913b4dd930b745af7e182a2a9806f)

## Reproduction:

If two characters have the same name, their underlying records (e.g. Medical and Security records) are shared. Basically, the record information for one character will show up for both themselves and other crew members with a matching name. This is seen when examining either of them with a Security or Medical HUD.

**Steps:**
1. Spawn in as a character with some name (Rita Starpattern) and some job (Assistant).
2. Commit suicide as Rita Starpattern the Assistant.
3. Respawn via the OOC commands as a new character with the same name. (E.g. Rita Starpattern, the Captain.)
4. Equip a HUD (either Security or Medical).
5. Examine yourself as Rita the Captain.
6. Notice that your `Rank` (aka: Job) is listed incorrectly as `Assistant`. (The job of your dead doppelganger).
    - **Note:** Criminal statuses are also shared, but require being examined by another character (because self examination doesn't show criminal statuses).
7. To check criminal statuses:
a. Flag the doppelganger (Assistant) as some type of criminal.
b. Suicide as the Captain, and then respawn as a 3rd character.
c. Wear a Sec-HUD and examine the Captain's body. They'll be flagged for the same crime as the Assistant.
    - I have no idea how Beepsky would react to an arrest warrant in this scenario.

Image demonstrating the results of steps 1 through 6:
![hud-records-issue](https://user-images.githubusercontent.com/5588048/224614288-90f1ccb1-d16c-4cfc-abd7-0a60eca3d935.png)

## Additional Notes:
I was not able to test this using two separate players that share a character name, but not the same CKey. So it's possible this may only occur when the doppelgangers share both a CKey AND a character name (meaning they're the same player).

On its own, this is likely a minor issue, since identical character names are rare. But it may be evidence of an underlying issue in how examine data is selected in HUDs.

**Comment by Sealed101** [comment]
needs repro for different ckeys just to be safe

**Comment by Mothblocks** [comment]
this is like one of the most well written repro steps in the 1.7k open issues we have though :(

**Comment by Sealed101** [comment]
i mostly sift through issues via mobile, and the comment was written via mobile as well. if anything it was mostly just a note to self/other issue jannies to test the issue out more thoroughly when possible 😄

**Comment by Sealed101** [comment]
yeah, the records fuck up with different ckeys as well (tested with a guest client)
i placed an arrest on the assistant and beepsky arrested us both, however the arrest balloon only appeared on the assistant

![image](https://user-images.githubusercontent.com/75863639/225299042-31aa62ac-a7ec-4cab-8e49-cdef86a02fd4.png)

however it would seem that it does create separate instances of records for respawns/same name characters

![image](https://user-images.githubusercontent.com/75863639/225303784-3903f492-cb9f-4afc-ad6c-83d95152c0c9.png)

and indeed, judging by the code, sechuds find perps by name, thus it only takes the first name on the list

https://github.com/tgstation/tgstation/blob/ba209adb358144454ea3ffeaa59e88d76cfad4b8/code/modules/mob/living/carbon/human/examine.dm#L368-L372
